mava/systems/ippo/networks.py

```diff
@@ -225,7 +245,21 @@ def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
         """
         # Add the observation network and an MLP network.
         policy_network = [
-            hk.nets.MLP(policy_layer_sizes, activation=jax.nn.relu),
+            hk.nets.MLP(
+                policy_layer_sizes,
+                activation=activation_function,
+                w_init=(
+                    lambda x: hk.initializers.Orthogonal(scale=jnp.sqrt(2))
```

Comment:
Do you need the lambda function here? Can you not use something like `w_init=hk.initializers.Orthogonal(scale=jnp.sqrt(2)) if orthogonal_initialisation else None`? Similar Q for the other cases. Also, if you can't as an argument, you can always assign the initialiser outside the network and use it as an argument later when defining the MLP.

Reply:
Or have lambda functions, but then reuse them where appropriate as @DriesSmit suggested.

Reply:
I created reusable `w_init_fn` and `b_init_fn` lambda functions. Please let me know your thoughts on this? :)